cuML/src/glm/ridge.h

```diff
@@ -65,94 +66,94 @@ void ridgeSolve(math_t *S, math_t *V, math_t *U, int n_rows, int n_cols,
 void ridgeSVD(math_t *A, int n_rows, int n_cols, math_t *b, math_t *alpha,
-		int n_alpha, math_t *w, cusolverDnHandle_t cusolverH,
-              cublasHandle_t cublasH, DeviceAllocator &mgr) {
-
+		int n_alpha, math_t *w, const cumlHandle_impl& handle) {
 	ASSERT(n_cols > 0,
 			"ridgeSVD: number of columns cannot be less than one");
 	ASSERT(n_rows > 1,
 			"ridgeSVD: number of rows cannot be less than two");
-
-	math_t *S, *V, *U;
+        auto allocator = handle.getDeviceAllocator();
+        auto stream = handle.getStream();
+        auto cusolverH = handle.getcusolverDnHandle();
+        auto cublasH = handle.getCublasHandle();
 
 	int U_len = n_rows * n_rows;
 	int V_len = n_cols * n_cols;
 
-	allocate(U, U_len);
-	allocate(V, V_len);
-	allocate(S, n_cols);
-
-	LinAlg::svdQR(A, n_rows, n_cols, S, U, V, true, true, cusolverH,
-                      cublasH, mgr);
-	ridgeSolve(S, V, U, n_rows, n_cols, b, alpha, n_alpha, w, cusolverH,
-			cublasH);
-
-	CUDA_CHECK(cudaFree(U));
-	CUDA_CHECK(cudaFree(V));
-	CUDA_CHECK(cudaFree(S));
+        device_buffer<math_t> U(allocator, stream, U_len);
+        device_buffer<math_t> V(allocator, stream, V_len);
+        device_buffer<math_t> S(allocator, stream, n_cols);
 
+	LinAlg::svdQR(A, n_rows, n_cols, S.data(), U.data(), V.data(), true, true,
+                      cusolverH, cublasH, allocator, stream);
+	ridgeSolve(S.data(), V.data(), U.data(), n_rows, n_cols, b, alpha, n_alpha, w,
+                   cusolverH, cublasH);
 }
 
 template<typename math_t>
 void ridgeEig(math_t *A, int n_rows, int n_cols, math_t *b, math_t *alpha,
-		int n_alpha, math_t *w, cusolverDnHandle_t cusolverH,
-              cublasHandle_t cublasH, DeviceAllocator &mgr) {
-
+		int n_alpha, math_t *w, const cumlHandle_impl& handle) {
 	ASSERT(n_cols > 1,
 			"ridgeEig: number of columns cannot be less than two");
 	ASSERT(n_rows > 1,
 			"ridgeEig: number of rows cannot be less than two");
-
-	math_t *S, *V, *U;
+        auto allocator = handle.getDeviceAllocator();
+        auto stream = handle.getStream();
+        auto cusolverH = handle.getcusolverDnHandle();
+        auto cublasH = handle.getCublasHandle();
 
 	int U_len = n_rows * n_cols;
 	int V_len = n_cols * n_cols;
 
-	allocate(U, U_len);
-	allocate(V, V_len);
-	allocate(S, n_cols);
-
-	LinAlg::svdEig(A, n_rows, n_cols, S, U, V, true, cublasH, cusolverH, mgr);
-	ridgeSolve(S, V, U, n_rows, n_cols, b, alpha, n_alpha, w, cusolverH,
-			cublasH);
+        device_buffer<math_t> U(allocator, stream, U_len);
+        device_buffer<math_t> V(allocator, stream, V_len);
+        device_buffer<math_t> S(allocator, stream, n_cols);
 
-	CUDA_CHECK(cudaFree(U));
-	CUDA_CHECK(cudaFree(V));
-	CUDA_CHECK(cudaFree(S));
+	LinAlg::svdEig(A, n_rows, n_cols, S.data(), U.data(), V.data(), true, cublasH, cusolverH,
+                       allocator, stream);
+	ridgeSolve(S.data(), V.data(), U.data(), n_rows, n_cols, b, alpha, n_alpha, w, cusolverH,
+                   cublasH);
```

Comment:
To allow better future compos-ability I think it would be better to explicitly pass a stream into ridgeSolve instead of implicitly doing so via cusolverH and cublasH (which might even be inconsistent).